Source/HTTP/Android/http_android.cpp

```diff
@@ -6,15 +6,40 @@
 #include "android_http_request.h"
 #include "android_platform_context.h"
 
+extern "C"
+{
+
+JNIEXPORT void JNICALL Java_com_xbox_httpclient_HttpClientRequest_OnRequestCompleted(JNIEnv* env, jobject instance, jlong call, jobject response)
+{
+    hc_call_handle_t sourceCall = reinterpret_cast<hc_call_handle_t>(call);
+    HttpRequest* sourceRequest = nullptr;
+
+    HCHttpCallGetContext(sourceCall, reinterpret_cast<void**>(&sourceRequest));
+
+    if (response == nullptr) 
+    {
+        CompleteAsync(sourceRequest->GetAsyncBlock(), E_FAIL, 0);
+    }
+    else 
+    {
+        HRESULT result = sourceRequest->ProcessResponse(sourceCall, response);
+        CompleteAsync(sourceRequest->GetAsyncBlock(), result, 0);
+    }
+
+    delete sourceRequest;
```

Comment:
>delete sourceRequest; [](start = 4, length = 21)

I'd like some advice for what to do about this; on windows it looks like there's a TODO to clean up the windows-specific type that's newed up. I could take ownership as a unique_ptr here and let that clean it up when the function goes out of scope if that's preferred.

I think I need to clean this up in the failure condition below as well.

Reply:
using a unique_ptr would be good to protect against weird edge cases

---
In reply to: [191911754](https://github.com/Microsoft/libHttpClient/pull/176#discussion_r191911754) [](ancestors = 191911754)